### ISSUE TITLE
Alt text composer is difficult to use for users with visual impairments

### ISSUE THREAD

**Issue by hkolbeck** [bluesky-social/social-app#991]
**Describe the bug**

Bug report from another user, who did not wish to share their contact info:

> some of us who can access it can't use the alt text function due to our vision issues that the function doesn't cater too
> For me to write alt text I have to use multiple other accessibility tools
> As it stands now an image on bsky can't be zoomed in on so I have to screenshot it leave the app and zoom in on the screenshot using the accessibility tools in my device
> So me doing alt text is a huge labour intensive thing

**To Reproduce**

1. Put on glasses with the wrong prescription
2. Try to use the alt text composer, describing details of the image at hand.

**Expected behavior**

Better support for zooming in on the image when composing alt text

**Details**

 - Platform: "Switch between browser and app", does not wish to say specifics
 - Platform version: Unsure
 - App version: Unsure

**Additional context**

I anticipate a push to reclassify this as a feature request, and I want anyone thinking that to consider that this report is that an important piece of the site's functionality does not work for some users. I recognize that your roadmap is chaos right now, and I want to acknowledge that you're doing some great work around alt text. Speaking to those who navigate your site using screen readers though, alt text is not the most pressing issue. They've suggested that you, and the users spending their time advocating around alt text, spend some time navigating the site exclusively using a screen reader.

If anyone is thinking that this affects only a small population, consider both that the set of those who use screen readers is larger than you think, and that the issues being discussed right now are a core reason *why* the population is as small as it is. I hope you'll seek out direct connection with the folks who rely on this functionality, and I'm happy to facilitate those connections if I can.

**Comment by ssg** [comment]
That's not even a visual impairment problem. Ability to examine the image while writing alt text is a necessary affordance. When the UI makes me go back and forth between the image and the text, it can be extremely cumbersome as if writing ALT text isn't unpopular enough.

An alternative could be to have an AI (on-device, like Apple Intelligence perhaps? not sure if it's possible at the moment) generate a description text for the image, and let us edit that. Then such an intricate UX may not be necessary for many cases, and can also popularize ALT texts as the process becomes way easier.

**Comment by hkolbeck** [comment]
@ssg I left the platform some time ago, is the image being described now not at all visible? I recall having frustration when trying to quote a post with an undescribed image to add a description.

I don't think on-device AI would be reliable enough even with editing encouraged, and the environmental impacts of remote would of course be considerable. In many related discussions people have wanted AI generation to work as it does on the meta platforms, applied to every image with no work by the poster. The generators on those sites are utter trash.

I do think AI could back two tools here:
- Let a post reader get a knowingly AI generated description with one click. That description would best include the text of the post or thread and the current topics on the platform in its generation because loss of context is behind a huge part of AI descriptions' failures, but that of course opens a can of worms in multiple respects.
- Let an image poster generate an AI description with a click and force review, with readers informed of AI usage and diff size from result.
